File: ios/StatexTodo/Task.swift
//
//  Task.swift
//  ToDoApp
//
//  Created by James Deng on 12/03/2016.
//  Copyright © 2016 Rewen. All rights reserved.
//

import UIKit

class Task: NSObject {
  let id: Int
  let title: String
  var completed: Bool
  
//  init(title: String, completed: Bool = false) {
//    self.title = title
//    self.completed = completed
//  }
  
  init(id: Int = -1, title: String, completed: Bool = false) {
    self.id = id
    self.title = title
    self.completed = completed
  }
  
  func dict() -> [String: AnyObject] {
    return self.dictionaryWithValuesForKeys(["title", "completed"])
  }
}

File: android/app/src/main/java/com/statextodo/Task.java
package com.statextodo;

/**
 * @author Junjun Deng 2016
 */
public class Task
{
	long id;
	String title;
	boolean completed;

	public Task(long id, String title, boolean completed)
	{
		this.id = id;
		this.title = title;
		this.completed = completed;
	}
}
